nano/node/node.cpp

```diff
@@ -3477,9 +3519,94 @@ void nano::active_transactions::update_difficulty (nano::block const & block_a)
 		uint64_t difficulty;
 		auto error (nano::work_validate (block_a, &difficulty));
 		assert (!error);
-		roots.modify (existing, [difficulty](nano::conflict_info & info_a) {
-			info_a.difficulty = difficulty;
-		});
+		if (difficulty > existing->difficulty)
+		{
+			roots.modify (existing, [difficulty](nano::conflict_info & info_a) {
+				info_a.difficulty = difficulty;
+			});
+			adjust_difficulty (block_a.hash ());
+		}
+	}
+}
+
+void nano::active_transactions::adjust_difficulty (nano::block_hash const & hash_a)
+{
+	assert (!mutex.try_lock ());
+	std::deque<std::pair<nano::block_hash, int64_t>> remaining_blocks;
+	remaining_blocks.push_back (std::make_pair (hash_a, 0));
```

Comment:
Rather than make a pair then copy/move it to the deque (which push_back will do) you can use remaining_blocks.emplace_back (previous, item.second + 1); which creates the pair directly inside the deque using _placement new_. The same can be done with the other uses in this function

Reply:
Excellent find!